### PR TITLE
Check in simple end-to-end test for covariant returns.

### PR DESCRIPTION
**Pull request by gafter**

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/Microsoft.CodeAnalysis.CSharp.Emit.UnitTests.csproj [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <PropertyGroup>
     <OutputType>Library</OutputType>
     <RootNamespace>Microsoft.CodeAnalysis.CSharp.UnitTests</RootNamespace>
-    <TargetFrameworks>netcoreapp3.1;net472</TargetFrameworks>
+    <TargetFrameworks>net5.0;net472</TargetFrameworks>
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/Microsoft.CodeAnalysis.CSharp.Emit.UnitTests.csproj:7]: Just to confirm, we're intentionally keeping our other test projects targeting netcoreapp3.1?

File: src/Compilers/CSharp/Test/Emit/Emit/CovariantReturnTests.cs [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Emit
+{
+    public class CovariantReturnTests : EmitMetadataTestBase
+    {
+        private static readonly MetadataReference CorelibraryWithCovariantReturnSupport;
+
+        static CovariantReturnTests()
+        {
+            if (new CovarantReturnRuntimeOnly().ShouldSkip)
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/Emit/CovariantReturnTests.cs:20]: >CovarantReturnRuntimeOnly [](start = 20, length = 25)

typo: Covariant

File: src/Compilers/CSharp/Test/Emit/Microsoft.CodeAnalysis.CSharp.Emit.UnitTests.csproj (hunk 1 above)
Comment (gafter) [reply src/Compilers/CSharp/Test/Emit/Microsoft.CodeAnalysis.CSharp.Emit.UnitTests.csproj:7]: I suspect it is @jaredpar's intention to switch them all at some point.

---
In reply to: [460321746](https://github.com/dotnet/roslyn/pull/46285#discussion_r460321746) [](ancestors = 460321746)

**Pull request merged**